File: src/main/java/upbrella/be/store/dto/response/SingleCurrentLocationStoreResponse.java
package upbrella.be.store.dto.response;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class SingleCurrentLocationStoreResponse {

    private long id;
    private String name;
    private boolean openStatus;
    private double latitude;
    private double longitude;
    private long rentableUmbrellasCount;

    public static SingleCurrentLocationStoreResponse fromStoreMeta(boolean openStatus, StoreMetaWithUmbrellaCount storeMetaWithUmbrellaCount) {

        return SingleCurrentLocationStoreResponse.builder()
                .id(storeMetaWithUmbrellaCount.getStoreMeta().getId())
                .name(storeMetaWithUmbrellaCount.getStoreMeta().getName())
                .openStatus(openStatus)
                .latitude(storeMetaWithUmbrellaCount.getStoreMeta().getLatitude())
                .longitude(storeMetaWithUmbrellaCount.getStoreMeta().getLongitude())
                .rentableUmbrellasCount(storeMetaWithUmbrellaCount.getRentableUmbrellasCount())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleCurrentLocationStoreResponse.kt
package upbrella.be.store.dto.response

data class SingleCurrentLocationStoreResponse(
    val id: Long?,
    val name: String,
    val openStatus: Boolean,
    val latitude: Double,
    val longitude: Double,
    val rentableUmbrellasCount: Long
) {
    companion object {
        fun fromStoreMeta(
            openStatus: Boolean,
            storeMetaWithUmbrellaCount: StoreMetaWithUmbrellaCount
        ): SingleCurrentLocationStoreResponse {
            return SingleCurrentLocationStoreResponse(
                id = storeMetaWithUmbrellaCount.storeMeta.id,
                name = storeMetaWithUmbrellaCount.storeMeta.name,
                openStatus = openStatus,
                latitude = storeMetaWithUmbrellaCount.storeMeta.latitude,
                longitude = storeMetaWithUmbrellaCount.storeMeta.longitude,
                rentableUmbrellasCount = storeMetaWithUmbrellaCount.rentableUmbrellasCount
            )
        }
    }
}
